test/IIS.FunctionalTests/Inprocess/StdOutRedirectionTests.cs

```diff
@@ -0,0 +1,208 @@
+
+            Assert.Contains("The specified framework 'Microsoft.NETCore.App', version '2.9.9' was not found.",
+                contents);
+            var count = 0;
+            foreach (var write in TestSink.Writes)
+            {
+                if (write.Message.Contains(
+                    "The specified framework 'Microsoft.NETCore.App', version '2.9.9' was not found."))
+                {
+                    count++;
+                }
+            }
+
+            if (deploymentParameters.ServerType == ServerType.IISExpress)
+            {
+                Assert.Equal(2, count);
+            }
+            else
+            {
+                Assert.Equal(1, count);
+            }
+        }
+
+        [ConditionalFact]
+        [SkipIfDebug]
+        public async Task EnableCoreHostTraceLogging_TwoLogFilesCreated()
+        {
+                var deploymentParameters =
+                    _fixture.GetBaseDeploymentParameters(_fixture.StartupExceptionWebsite, publish: true);
+                deploymentParameters.EnvironmentVariables["COREHOST_TRACE"] = "1";
+                deploymentParameters.GracefulShutdown = true;
+
+                deploymentParameters.EnableLogging(_logFolderPath);
+
+                var deploymentResult = await DeployAsync(deploymentParameters);
+
+                var response = await deploymentResult.HttpClient.GetAsync("/");
+                Assert.False(response.IsSuccessStatusCode);
+
+                StopServer();
+
+                var filesInDirectory = Directory.GetFiles(_logFolderPath);
+                Assert.Equal(2, filesInDirectory.Length);
+                foreach (var file in filesInDirectory)
+                {
+                    var contents = File.ReadAllText(file);
+                    Assert.Contains("Invoked hostfxr", contents);
+                }
+        }
+
+        [ConditionalTheory]
+        [SkipIfDebug]
+        [InlineData("CheckLargeStdErrWrites")]
+        [InlineData("CheckLargeStdOutWrites")]
+        [InlineData("CheckOversizedStdErrWrites")]
+        [InlineData("CheckOversizedStdOutWrites")]
+        public async Task EnableCoreHostTraceLogging_PipeCaptureNativeLogs(string path)
+        {
+            var deploymentParameters = _fixture.GetBaseDeploymentParameters(_fixture.StartupExceptionWebsite, publish: true);
+            deploymentParameters.EnvironmentVariables["COREHOST_TRACE"] = "1";
+            deploymentParameters.WebConfigBasedEnvironmentVariables["ASPNETCORE_INPROCESS_STARTUP_VALUE"] = path;
+            deploymentParameters.GracefulShutdown = true;
+
+            var deploymentResult = await DeployAsync(deploymentParameters);
+
+            var response = await deploymentResult.HttpClient.GetAsync("/");
+
+            Assert.False(response.IsSuccessStatusCode);
+
+            StopServer();
+
+            Assert.Contains(TestSink.Writes, context => context.Message.Contains("Invoked hostfxr"));
```

Comment:
Let's not assert against TestSink at all in this class. We are interested in either getting text in event log or log file.